crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -1255,60 +1250,101 @@ impl CumulocityConverter {
             }
         }
     }
-}
 
-async fn publish_operation_status(
-    json_response: &str,
-    http_proxy: &mut C8YHttpProxy,
-    device_id: String,
-) -> Result<Vec<Message>, CumulocityMapperError> {
-    let response = SoftwareUpdateResponse::from_json(json_response)?;
-    let topic = C8yTopic::SmartRestResponse.to_topic()?;
-    match response.status() {
-        OperationStatus::Executing => {
-            let smartrest_set_operation_status =
-                SmartRestSetOperationToExecuting::from_thin_edge_json(response)?.to_smartrest()?;
-            Ok(vec![Message::new(&topic, smartrest_set_operation_status)])
+    async fn register_software_list_operation(
```

Comment:
Indeed, it would be good to have operation specific modules or even actor extensions. However, I would prefer to address that later once both software list and software update fully ported to support child devices.